client/js/tracking.js

```diff
@@ -2,23 +2,38 @@
 
 import { on } from './util';
 
-export default {
-  init: () => {
-    // GA events
-    on('body', 'click', '[data-track-event]', ({ target }) => {
-      const el = target.closest('[data-track-event]');
-      trackGaEvent(JSON.parse(el.getAttribute('data-track-event')));
-    });
+const maybeTrackEvent = (hasAnalytics) => {
+  if (hasAnalytics) {
+    const actuallyTrackEvent = ({category, action, label}) => {
+      ga('send', {
+        hitType: 'event',
+        eventCategory: category,
+        eventAction: action,
+        eventLabel: label
+      });
+    };
+    return actuallyTrackEvent;
+  } else {
+    const noop = () => {};
+    return noop;
   }
 };
 
-function trackOutboundLink(url) {
-  ga('send', 'event', 'outbound', 'click', url, {
-    'transport': 'beacon',
-    'hitCallback': () => {
-      document.location = url;
-    }
-  });
+const maybeTrackOutboundLinks = (hasAnalytics) => {
+  if (hasAnalytics) {
+    const actuallyTrackLinks = (url) => {
+      ga('send', 'event', 'outbound', 'click', url, {
+        'transport': 'beacon',
+        'hitCallback': () => {
+          document.location = url;
+        }
+      });
+    };
+    return actuallyTrackLinks;
+  } else {
+    const noop = () => {};
```

Comment:
I guess you could set `noop` outside all of these functions instead of creating it inside each?